src/ray/raylet/task_dependency_manager.h

```diff
@@ -55,16 +58,39 @@ class TaskDependencyManager {
   /// \param required_objects The objects required by the task.
   /// \return Whether all of the given dependencies for the given task are
   /// local.
-  bool SubscribeDependencies(const TaskID &task_id,
-                             const std::vector<ObjectID> &required_objects);
+  bool SubscribeGetDependencies(const TaskID &task_id,
+                                const std::vector<ObjectID> &required_objects);
 
-  /// Unsubscribe from the object dependencies required by this task. If the
-  /// objects were remote and are no longer required by any subscribed task,
-  /// then they will be canceled.
+  /// Subscribe to object depedencies required by the worker. This should be called for
+  /// ray.wait calls during task execution.
+  ///
+  /// The TaskDependencyManager will track all remote dependencies until the
+  /// dependencies are local, or until UnsubscribeWaitDependencies is called
+  /// with the same worker ID, whichever occurs first. Remote dependencies will
+  /// be requested.  This method may be called multiple times per worker on the
+  /// same objects.
   ///
-  /// \param task_id The ID of the task whose dependencies to unsubscribe from.
+  /// \param worker_id The ID of the worker that called `ray.wait`.
+  /// \param required_objects The objects required by the worker.
+  /// \return Void.
+  void SubscribeWaitDependencies(const WorkerID &worker_id,
+                                 const std::vector<ObjectID> &required_objects);
+
+  /// Unsubscribe from the object dependencies required by this task through the task
+  /// arguments or `ray.get`. If the objects were remote and are no longer required by any
+  /// subscribed task, then they will be canceled.
+  ///
+  /// \param task_id The ID of the task whose dependencies we should unsubscribe from.
   /// \return Whether the task was subscribed before.
-  bool UnsubscribeDependencies(const TaskID &task_id);
+  bool UnsubscribeGetDependencies(const TaskID &task_id);
+
+  /// Unsubscribe from the object dependencies required by this worker through `ray.wait`.
+  /// If the objects were remote and are no longer required by any subscribed task, then
+  /// they will be canceled.
+  ///
+  /// \param worker_id The ID of the worker whose dependencies we should unsubscribe from.
+  /// \return The objects that the worker was waiting on.
+  void UnsubscribeWaitDependencies(const WorkerID &worker_id);
```

Comment:
It would be nice to not have separate APIs for Get and Wait deps.

Reply:
Yeah, I agree it's not great...but the semantics are quite different and I'm not really sure how/whether to unify them.